test/visual/test.ts

```diff
@@ -68,6 +97,53 @@ async function testReady(test: StoryDecorator, retry = 0): Promise<void> {
     );
 }
 
+/**
+ * Ensures component rendering is stable before taking screenshots
+ * Waits for layout, animations, and pending renders to complete
+ *
+ * @param root - The root element containing the component under test
+ */
+async function ensureComponentStable(root: Element): Promise<void> {
+    // Force a layout computation
+    root.getBoundingClientRect();
+
+    // First frame to process attribute changes
+    await nextFrame();
+
+    // Wait for all animations to complete
+    try {
+        const elements = Array.from(root.querySelectorAll('*'));
+        const animations = elements.flatMap((el) => {
```

Comment:
Instead of querying all elements, you can pass `{subtree: true}` to `el.getAnimations({subtree: true})`. This would catch animations from pseudo-elements and the shadow DOM. In the current case, the shadow DOM is not being traversed.